extensions/cornerstone/src/init.tsx

```diff
@@ -310,37 +316,73 @@ export default async function init({
-  const activeToasts = new Map();
-
-  // eventTarget.addEventListener(EVENTS.WEB_WORKER_PROGRESS, ({ detail }) => {
-  //   const { progress, type, id } = detail;
-
-  //   const cacheKey = `${type}-${id}`;
-  //   if (progress === 0 && !activeToasts.has(cacheKey)) {
-  //     const progressPromise = new Promise((resolve, reject) => {
-  //       activeToasts.set(cacheKey, { resolve, reject });
-  //     });
-
-  //     uiNotificationService.show({
-  //       id: cacheKey,
-  //       title: `${type}`,
-  //       message: `${type}: ${progress}%`,
-  //       autoClose: false,
-  //       promise: progressPromise,
-  //       promiseMessages: {
-  //         loading: `Computing...`,
-  //         success: `Completed successfully`,
-  //         error: 'Web Worker failed',
-  //       },
-  //     });
-  //   } else {
-  //     if (progress === 100) {
-  //       const { resolve } = activeToasts.get(cacheKey);
-  //       resolve({ progress, type });
-  //       activeToasts.delete(cacheKey);
-  //     }
-  //   }
-  // });
+  // Use a single map to track all active worker tasks
+  const activeWorkerTasks = new Map();
+
+  // Create a normalized task key that doesn't include the random ID
+  // This helps us identify and deduplicate the same type of task
+  const getNormalizedTaskKey = type => {
+    return `worker-task-${type.toLowerCase().replace(/\s+/g, '-')}`;
+  };
+
+  eventTarget.addEventListener(EVENTS.WEB_WORKER_PROGRESS, ({ detail }) => {
+    const { progress, type, id } = detail;
+
+    // Skip notifications for compute statistics
+    if (type === cornerstoneTools.Enums.WorkerTypes.COMPUTE_STATISTICS) {
+      return;
+    }
+
+    const normalizedKey = getNormalizedTaskKey(type);
+
+    if (progress === 0) {
+      // Check if we're already tracking a task of this type
+      if (!activeWorkerTasks.has(normalizedKey)) {
+        const progressPromise = new Promise((resolve, reject) => {
+          activeWorkerTasks.set(normalizedKey, {
+            resolve,
+            reject,
+            originalId: id,
+            type,
+          });
+        });
+
+        uiNotificationService.show({
+          id: normalizedKey, // Use the normalized key as ID for better deduplication
+          title: `${type}`,
+          message: `Computing...`,
+          autoClose: false,
+          allowDuplicates: false,
+          deduplicationInterval: 60000, // 60 seconds - prevent frequent notifications of same type
+          promise: progressPromise,
+          promiseMessages: {
+            loading: `Computing...`,
+            success: `Completed successfully`,
+            error: 'Web Worker failed',
+          },
+        });
+      } else {
+        // Already tracking this type of task, just let it continue
+        console.debug(`Already tracking a "${type}" task, skipping duplicate notification`);
+      }
+    }
+    // Task completed
+    else if (progress === 100) {
```

Comment:
stupid question... what happens if the worker errors out? Will the task get removed from the `activeWorkerTasks`? Maybe we don't care, but I wanted for us to be sure. Furthermore, does the promise need to be rejected in that case too?